website/src/content/docs/guides/capture/capture_resume.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Capture Resume
+description: Resuming a Crashed Capture
+---
+
+A capture campaign can easily take anywhere from a couple of minutes to several
+months.  A crash may occur for many reasons such as a power outage.  When a
+crash happens the generated inputs by attack point iterators would not be
+completely balanced.  We build our tooling to be resistant to crashes and thus
+resumable.  We first save all inputs of the algorithm of interest and then
+iterate those while automatically saving progress.
+
+## Resume Capture Overview
+
+First we save all we create all input values.  Save those.  Then we load and
+iterate and in case of an error or a crash we just reload where we stopped.
+Let us go through the code with the concrete case of AES128.
+
+### Save Inputs
+
+In the last guide we created iterators over inputs.  We can accumulate all
+values in arrays `keys` and `plaintexts` where `keys.shape = (65_536, 16)` and
+`plaintexts.shape = (65_536, 16)`.
+
+```python
+from scaaml.io.resume_kti import create_resume_kti
+
+# Save all encryption parameters ahead before the capture campaign starts.
+# If `kt_filename` exists this does nothing.
+create_resume_kti(
+  parameters={"keys": keys, "texts": texts},
+  shard_length=32,
+  kt_filename="parameters_tuples.txt",
+  progress_filename="progress_tuples.txt",
+)
+```
+
+-   We save measurements in files called shards.  Each of which contains
+    `shard_length` examples.  An error could happen while saving the file.  We
+    thus restart at the lowest multiple of `shard_length` to ensure that data
+    have been saved.  One can set `shard_length=1` so that we do not repeat
+    successful captures.
+-   Note that it is not necessary for the `keys` and `plaintexts` to have the
+    same length.  The only condition we require is that there is the same
+    number of those.  That is we need to `assert keys.shape[0] ==
+    plaintexts.shape[0]`.
+-   If we wanted to save more values in `parameters` we could.  The only
+    condition is that all of them have the same `len`.
+
```

Comment:
Might be worth adding a line or 2 about what happens if the input generators are infinite